query/expression.go

```diff
@@ -116,6 +118,8 @@ func (expr stringExpression) Evaluate(context EvaluationContext) (value, error)
 }
 
 func (expr *metricFetchExpression) Evaluate(context EvaluationContext) (value, error) {
+	// Record the duration of this request in the profiler.
+	defer context.Profiler.Add("metricFetchExpression.Evaluate()", inspect.Now())
```

Comment:
another way to do this is i think, making an `Add()` function that returns a function. this will remove the need to have `inspect.Now()` as a parameter.

`defer context.Profile.Add("foo") -> returns func()`